orchestrator/relayer/src/find_latest_valset.rs

```diff
@@ -81,20 +81,19 @@ pub async fn find_latest_valset(
 /// The other (and far worse) way a disagreement here could occur is if validators are colluding to steal
 /// funds from the Gravity contract and have submitted a highjacking update. If slashing for off Cosmos chain
 /// Ethereum signatures is implemented you would put that handler here.
-fn check_if_valsets_differ(cosmos_valset: Option<Valset>, ethereum_valset: &Valset) {
+fn check_if_valsets_differ(
+    cosmos_valset: Option<Valset>,
+    ethereum_valset: &Valset,
+) -> Result<(), GravityError> {
     if cosmos_valset.is_none() && ethereum_valset.nonce == 0 {
         // bootstrapping case
-        return;
+        return Ok(());
     } else if cosmos_valset.is_none() {
         error!("Cosmos does not have a valset for nonce {} but that is the one on the Ethereum chain! Possible bridge highjacking!", ethereum_valset.nonce);
-        return;
+        return Ok(());
     }
     let cosmos_valset = cosmos_valset.unwrap();
     if cosmos_valset != *ethereum_valset {
-        // if this is not true then we have a logic error on the Cosmos chain
-        // or with our Ethereum search
-        assert_eq!(cosmos_valset.nonce, ethereum_valset.nonce);
```

Comment:
If this condition really is programmer error we may want to leave this assertion in.